test/deprecation_exec.jl

```diff
@@ -0,0 +1,69 @@
+using Base.Test
+
+module DeprecationTests # to test @deprecate
+    f() = true
+
+    # test the Symbol path of @deprecate
+    @deprecate f1 f
+    @deprecate f2 f false # test that f2 is not exported
+
+    # test the Expr path of @deprecate
+    @deprecate f3() f()
+    @deprecate f4() f() false # test that f4 is not exported
+    @deprecate f5(x::T) where T f()
+
+    # test deprecation of a constructor
+    struct A{T} end
+    @deprecate A{T}(x::S) where {T, S} f()
+
+    # test that @deprecate_moved can be overridden by an import
+    Base.@deprecate_moved foo1234 "Foo"
+    Base.@deprecate_moved bar "Bar" false
+end # module
+module Foo1234
+    export foo1234
+    foo1234(x) = x+1
+end
+
+# issue #21972
+struct T21972
+    @noinline function T21972()
+        Base.depwarn("something", :T21972)
+        new()
+    end
+end
+
+@testset "@deprecate" begin
```

Comment:
hm, both this testset and `in anonymous at <missing>` are printing here and escaping both the runtests.jl printing and `@test_warn` capturing output